scripts/sentiment_analysis/sentiment_analysis_cnn.py

```diff
@@ -0,0 +1,319 @@
+train_dataset, valid_dataset = nlp.data.train_valid_split(train_dataset, args.valid_ratio)
+
+all_token = []
+for line in train_dataset:
+    all_token.extend(tokenizer(line[0]))
+vocab = nlp.Vocab(nlp.data.count_tokens(all_token))
+vocab.set_embedding(nlp.embedding.create('Word2Vec', source='GoogleNews-vectors-negative300'))
+print(len(vocab))
+net = SentimentNet(dropout=args.dropout, vocab_size=len(vocab), use_multichannel=True)
+
+print('Tokenize using spaCy...')
+
+# Dataset preprocessing
+def preprocess(x):
+    data, label = x
+    label = int(label > 5)
+    data = vocab[length_clip(tokenizer(data))]
+    return data, label
+
+def get_length(x):
+    return float(len(x[0]))
+
+def preprocess_dataset(dataset):
+    start = time.time()
+    pool = mp.Pool(2)
+    dataset = gluon.data.SimpleDataset(pool.map(preprocess, dataset))
+    lengths = gluon.data.SimpleDataset(pool.map(get_length, dataset))
+    end = time.time()
+    print('Done! Tokenizing Time={:.2f}s, #Sentences={}'.format(end - start, len(dataset)))
+    return dataset, lengths
+
+# Preprocess the dataset
+train_dataset, train_data_lengths = preprocess_dataset(train_dataset)
+valid_dataset, valid_data_lengths = preprocess_dataset(valid_dataset)
+test_dataset, test_data_lengths = preprocess_dataset(test_dataset)
+
+# Construct the DataLoader. Pad data and stack label
+batchify_fn = nlp.data.batchify.Tuple(nlp.data.batchify.Pad(axis=0, ret_length=True),
+                                      nlp.data.batchify.Stack(dtype='float32'))
+if args.bucket_type is None:
+    print('Bucketing strategy is not used!')
+    train_dataloader = DataLoader(dataset=train_dataset,
+                                  batch_size=args.batch_size,
+                                  shuffle=True,
+                                  batchify_fn=batchify_fn)
+else:
+    if args.bucket_type == 'fixed':
+        print('Use FixedBucketSampler')
+        batch_sampler = nlp.data.FixedBucketSampler(train_data_lengths,
+                                                    batch_size=args.batch_size,
+                                                    num_buckets=args.bucket_num,
+                                                    ratio=args.bucket_ratio,
+                                                    shuffle=True)
+        print(batch_sampler.stats())
+    elif args.bucket_type == 'sorted':
+        print('Use SortedBucketSampler')
+        batch_sampler = nlp.data.SortedBucketSampler(train_data_lengths,
+                                                     batch_size=args.batch_size,
+                                                     mult=args.bucket_mult,
+                                                     shuffle=True)
+    else:
+        raise NotImplementedError
+    train_dataloader = DataLoader(dataset=train_dataset,
+                                  batch_sampler=batch_sampler,
+                                  batchify_fn=batchify_fn)
```

Comment:
If you plan to provide bucket_type options, make sure that when users set 'fixed' or 'sorted', this script still produces correct results.